developer/workflow/creating-a-workflow-activity-from-a-windows-powershell-cmdlet.md

```diff
@@ -1,118 +1,122 @@
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Management.Automation;
+using System.Collections.ObjectModel;
+using Microsoft.PowerShell.Activities;
+
+namespace MakeActivity
+{
+    class Program
+    {
+        static void Main(string[] args)
+
+        {
+            StreamWriter CodeFile = new StreamWriter("c:\\\\testmodule\\codefile.cs");
+            Collection<PSModuleInfo> ModuleInfos = new Collection<PSModuleInfo> { };
+            PSModuleInfo ModuleInfo;
+            string ActivityCode = "";
+            string ModulePath = "";
+            Console.Write("Type the full path and filename of the module to process:");
+            ModulePath = Console.ReadLine();
+
+            PowerShell ps = PowerShell.Create();
+
+            ps.AddCommand("Import-Module").AddArgument(ModulePath);
+            ps.AddParameter("PassThru");
+            ModuleInfos = ps.Invoke<PSModuleInfo>();
+
+            ModuleInfo = (PSModuleInfo)ModuleInfos[0];
+
+            ActivityCode = ActivityGenerator.GenerateFromModuleInfo(ModuleInfo, "MyNamespace").First<String>();
+
+           CodeFile.Write(ActivityCode);
+            Console.ReadLine();
+
+        }
+    }
+}
+
+```
+
+## Example
+
+ The following example demonstrates how to generate C# code for an activity from a Windows PowerShell cmdlet.
+
+```csharp
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Management.Automation;
+using System.Collections.ObjectModel;
+using Microsoft.PowerShell.Activities;
+
+namespace MakeActivity
+{
+    class Program
+    {
+        static void Main(string[] args)
+
+        {
+            StreamWriter CodeFile = new StreamWriter("c:\\\\testmodule\\codefile.cs");
+            Collection<PSModuleInfo> ModuleInfos = new Collection<PSModuleInfo> { };
+            PSModuleInfo ModuleInfo;
+            Collection<CommandInfo> CommandInfos = new Collection<CommandInfo> { };
+            CommandInfo MyCommand;
+            string ActivityCode = "";
+            string ModulePath = "";
+            Console.Write("Type the full path and filename of the module to process:");
+            ModulePath = Console.ReadLine();
+
+            PowerShell ps = PowerShell.Create();
+
+            ps.AddCommand("Import-Module").AddArgument(ModulePath);
+            ps.AddParameter("PassThru");
+            ModuleInfos = ps.Invoke<PSModuleInfo>();
+
+            ModuleInfo = (PSModuleInfo)ModuleInfos[0];
+
+            ps.AddCommand("Get-Command").AddArgument(ModuleInfo);
+            CommandInfos = ps.Invoke<CommandInfo>();
+
+            MyCommand = CommandInfos[0];
+
+            ActivityCode = ActivityGenerator.GenerateFromCommandInfo(MyCommand, "MyNamespace");
+
+           CodeFile.Write(ActivityCode);
+            Console.ReadLine();
+
+        }
+    }
+}
+
+```


```

Comment:
control character

Reply:
This was an issue with my code fence logic and is fixed